docs/source/graphbolt/ondisk-dataset.rst

```diff
@@ -0,0 +1,829 @@
+        type: <string>
+        name: <string>
+        format: <string>
+        in_memory: <bool>
+        path: <string>
+    tasks:
+      - name: <string>
+        num_classes: <int>
+        train_set:
+          - type: <string>
+            data:
+              - name: <string>
+                format: <string>
+                in_memory: <bool>
+                path: <string>
+              - name: <string>
+                format: <string>
+                in_memory: <bool>
+                path: <string>
+        validation_set:
+          - type: <string>
+            data:
+              - name: <string>
+                format: <string>
+                in_memory: <bool>
+                path: <string>
+              - name: <string>
+                format: <string>
+                in_memory: <bool>
+                path: <string>
+        test_set:
+          - type: <string>
+            data:
+              - name: <string>
+                format: <string>
+                in_memory: <bool>
+                path: <string>
+              - name: <string>
+                format: <string>
+                in_memory: <bool>
+                path: <string>
+
+``dataset_name``
+^^^^^^^^^^^^^^^
+
+The ``dataset_name`` field is used to specify the name of the dataset. It is
+user-defined.
+
+``graph``
+^^^^^^^^
+
+The ``graph`` field is used to specify the graph structure. It has two fields:
+``nodes`` and ``edges``.
+
+ - ``nodes``: ``list``
+
+   The ``nodes`` field is used to specify the number of nodes for each node type.
+   It is a list of ``node`` objects. Each ``node`` object has two fields: ``type``
+   and ``num``.
+    - ``type``: ``string``, optional
+
+      The ``type`` field is used to specify the node type. It is ``null`` for
+      homogeneous graphs. For heterogeneous graphs, it is the node type.
+    - ``num``: ``int``
+
+      The ``num`` field is used to specify the number of nodes for the node type.
+      It is mandatory for both homogeneous graphs and heterogeneous graphs.
+
+  - ``edges``: ``list``
+
+    The ``edges`` field is used to specify the edges. It is a list of ``edge``
+    objects. Each ``edge`` object has three fields: ``type``, ``format`` and
+    ``path``.
+    - ``type``: ``string``, optional
+
+      The ``type`` field is used to specify the edge type. It is ``null`` for
+      homogeneous graphs. For heterogeneous graphs, it is the edge type.
+    - ``format``: ``string``
+
+      The ``format`` field is used to specify the format of the edge data. It can
+      only be ``csv`` for now.
+    - ``path``: ``string``
+
+      The ``path`` field is used to specify the path of the edge data. It is
+      relative to the directory of ``metadata.yaml`` file.
+
+
+``feature_data``
+^^^^^^^^^^^^^^^
+
+The ``feature_data`` field is used to specify the feature data. It is a list of
+``feature`` objects. Each ``feature`` object has five fields: ``domain``, ``type``,
+``name``, ``format`` and ``path``.
```

Comment:
also mention metadata here?

Reply:
it's not ready yet. i will update the doc once it's supported.